File: androidApp/src/main/java/com/alextos/darts/android/common/presentation/components/TurnItem.kt
package com.alextos.darts.android.common.presentation.components

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import com.alextos.darts.android.common.presentation.extensions.rowColor
import com.alextos.darts.android.common.presentation.extensions.textColor
import com.alextos.darts.core.domain.model.Turn

@Composable
fun TurnItem(
    turn: Turn,
    useTurnColors: Boolean,
    onSelect: () -> Unit
) {
    val textColor = turn.textColor()
    val backgroundColor = if (useTurnColors) turn.rowColor() else Color.Transparent
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .background(backgroundColor)
            .clickable { onSelect() }
            .padding(16.dp)
    ) {
        turn.shots.forEach {
            Cell(
                modifier = Modifier.weight(1f),
                text = it.sector.valueString(),
                textColor = textColor
            )
        }
        (0 until turn.shotsLeft()).forEach { _ ->
            Cell(
                modifier = Modifier.weight(1f),
                text = "",
                textColor = textColor
            )
        }
        Cell(
            modifier = Modifier.weight(1f),
            text = turn.score().toString(),
            textColor = textColor
        )
        Cell(
            modifier = Modifier.weight(1f),
            text = turn.leftAfter.toString(),
            textColor = textColor
        )
    }
}

File: iosApp/iosApp/Library/Components/TurnItem.swift
import SwiftUI
import shared

struct TurnItem: View {
	let turn: Turn
	let shotsLeft: Int
	let useTurnColors: Bool
	let onSelect: () -> Void
	
	var body: some View {
		HStack(spacing: 0) {
			ForEach(turn.shots) { shot in
				Cell(text: shot.sector.valueString())
			}
			if shotsLeft == 3 {
				ForEach(0 ..< 3) { _ in
					Cell(text: " ")
				}
			} else if shotsLeft == 2 {
				ForEach(0 ..< 2) { _ in
					Cell(text: " ")
				}
			} else if shotsLeft == 1 {
				ForEach(0 ..< 1) { _ in
					Cell(text: " ")
				}
			}

			Cell(text: "\(turn.score())")
			Cell(text: "\(turn.leftAfter)")
		}
		.contentShape(Rectangle())
		.onTapGesture {
			onSelect()
		}
		.background(useTurnColors ? turn.rowColor : .clear)
		.frame(maxWidth: .infinity)
	}
}
